psl/psl-core/src/validate/datasource_loader.rs

```diff
@@ -163,6 +163,24 @@ fn lift_datasource(
         None => (None, None),
     };
 
+    if let Some((shadow_url, _)) = shadow_database_url.clone() {
+        if let (Some(direct_url), Some(direct_url_span)) = (direct_url.clone(), direct_url_span) {
+            if shadow_url == direct_url {
+                diagnostics.push_error(DatamodelError::new_shadow_database_is_same_as_direct_url_error(
+                    source_name,
+                    direct_url_span,
+                ));
+            }
+        }
+
+        if shadow_url == url {
+            diagnostics.push_error(DatamodelError::new_shadow_database_is_same_as_main_url_error(
+                source_name,
+                url_span,
+            ));
+        }
+    }
+
```

Comment:
```suggestion
    if let Some((shadow_url, _)) = &shadow_database_url {
        if let (Some(direct_url), Some(direct_url_span)) = (&direct_url, direct_url_span) {
            if shadow_url == direct_url {
                diagnostics.push_error(DatamodelError::new_shadow_database_is_same_as_direct_url_error(
                    source_name,
                    direct_url_span,
                ));
            }
        }

        if shadow_url == &url {
            diagnostics.push_error(DatamodelError::new_shadow_database_is_same_as_main_url_error(
                source_name,
                url_span,
            ));
        }
    }

```

We don't need these clones, they're pretty expensive; we can just take references :)

Reply:
Ah gotcha, as you can see I'm still new to this Rust thing people keep talking about 😄